File: INTRODUCCIONPOO/EJERCICIO3/Ejercicio3.java
package PRACTICA1.INTRODUCCIONPOO.EJERCICIO3;

public class Ejercicio3 {
    public static class Coche {
        private String marca;
        private String modelo;
        private int velocidad;
    
        public Coche(String marca, String modelo, int velocidad) {
            this.marca = marca;
            this.modelo = modelo;
            this.velocidad = velocidad;
        }
    
        public void acelerar() {
            velocidad += 10;
            System.out.println(marca + " " + modelo + " acelera. Nueva velocidad: " + velocidad + " km/h");
        }
    
        public void frenar() {
            velocidad -= 5;
            if (velocidad < 0) {
                velocidad = 0;
            }
            System.out.println(marca + " " + modelo + " frena. Nueva velocidad: " + velocidad + " km/h");
        }
    
        public void mostrarVelocidad() {
            System.out.println("El coche " + marca + " " + modelo + " va a " + velocidad + " km/h");
        }
    
        public static void main(String[] args) {
            Coche coche1 = new Coche("Toyota", "Corolla", 50);
            Coche coche2 = new Coche("Ford", "Focus", 30);
    
            coche1.acelerar();
            coche1.frenar();
            coche1.mostrarVelocidad();
    
            coche2.acelerar();
            coche2.frenar();
            coche2.mostrarVelocidad();
        }
    }    
}

File: INTRODUCCIONPOO/EJERCICIO3/Ejercicio3.py
class Coche:
    def __init__(self, marca, modelo, velocidad):
        self.__marca = marca
        self.__modelo = modelo
        self.__velocidad = velocidad

    def acelerar(self):
        self.__velocidad += 10
        print(f"{self.__marca} {self.__modelo} acelera. Nueva velocidad: {self.__velocidad} km/h")

    def frenar(self):
        self.__velocidad -= 5
        if self.__velocidad < 0:
            self.__velocidad = 0
        print(f"{self.__marca} {self.__modelo} frena. Nueva velocidad: {self.__velocidad} km/h")

    def mostrar_velocidad(self):
        print(f"El coche {self.__marca} {self.__modelo} va a {self.__velocidad} km/h")

coche1 = Coche("Toyota", "Corolla", 50)
coche2 = Coche("Ford", "Focus", 30)

coche1.acelerar()
coche1.frenar()
coche1.mostrar_velocidad()

coche2.acelerar()
coche2.frenar()
coche2.mostrar_velocidad()